pulsar-broker/src/test/java/org/apache/pulsar/broker/service/TopicOwnerTest.java

```diff
@@ -240,6 +245,28 @@ private void spyZookeeperToDisconnectAfterPersist(ZooKeeper zooKeeper, RequestMa
         }).when(spyZooKeeperServer).submitRequest(any(Request.class));
     }
 
+    private MutableBoolean listenNamespaceBundleLoading(PulsarService broker, NamespaceBundle namespaceBundle) {
+        NamespaceService namespaceService = broker.getNamespaceService();
+        final MutableBoolean loaded = new MutableBoolean();
+        namespaceService.addNamespaceBundleOwnershipListener(new NamespaceBundleOwnershipListener() {
```

Comment:
If call listenNamespaceBundleLoading multiple times, it will register multiple listeners to the namespace service, is this an expected behavior for your test? In my opinion, we only need to add one listener for the same purpose?

And the MutableBoolean will be changed during the test execution, I noticed the MutableBoolean is used for asserting, this will introduce flaky tests.

Reply:
I am not sure I understand your concern correctly. I try to answer with following. Correct me if mistake happens.

* The returned `MutableBoolean` is one per listening. It is used in assertion after all procedures done. So if assertion fails, it is a bug either production or test scope.
* Every listening is serving for one bundle(eg. `namespaceBundle` in parameter list). Multiple listening return different `MutableBoolean` for combination of `broker` and `namespaceBundle`. There is no sharing.

> If call listenNamespaceBundleLoading multiple times, it will register multiple listeners to the namespace service, is this an expected behavior for your test?

There is no multiple listening for same combination of `broker` and `namespaceBundle` in test code.